environment/container/dockerengine.go

```diff
@@ -231,14 +233,26 @@ func (e *dockerEngine) CopyDirsIntoImage(image, newImageName string, paths map[s
 	return nil
 }
 
-// CopyDirsIntoContainer copies some directories into a container
-func (e *dockerEngine) CopyDirsIntoContainer(containerID string, paths map[string]string) (err error) {
+// CopyDataIntoContainer copies files or directories into a container
+func (e *dockerEngine) CopyDataIntoContainer(containerID string, paths map[string]string) (err error) {
 	for sp, dp := range paths {
-		err = copyDirToContainer(e.ctx, e.cli, containerID, sp, dp)
+        fi, err := os.Stat(sp)
 		if err != nil {
-			return fmt.Errorf("container data copy failed for image '%s' with volume %s:%s . Error: %w", containerID, sp, dp, err)
+			return fmt.Errorf("failed to stat the source path '%s' . Error: %w", sp, err)
 		}
-	}
+		if fi.Mode().IsDir() {
+            err = copyDirToContainer(e.ctx, e.cli, containerID, sp, dp)
+            if err != nil {
+                return fmt.Errorf("container data copy failed for image '%s' with volume %s:%s . Error: %w", containerID, sp, dp, err)
+            }
+        } else {
+            newDestPath := filepath.Join(dp, fi.Name())
+            err = copyDirToContainer(e.ctx, e.cli, containerID, sp, newDestPath)
```

Comment:
If `copyDirToContainer` is being used to copy files then we should rename it as well.